llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/TileAndDecomposeAttention.cpp

```diff
@@ -420,6 +426,61 @@ tileAndDecomposeAttention(IREE::LinalgExt::AttentionOp attnOp,
                         headDimension, loc, rewriter);
 
   attnOp.getResults()[0].replaceAllUsesWith(loopNest.results[0]);
+  ops.push_back(tiledAttentionOp);
+  return ops;
+}
+
+/// Decompose iree_linalg_ext.tiled_attention op.
+/// TODO: Adopt decomposeOperation with this.
+static void
+decomposeTiledAttention(IREE::LinalgExt::TiledAttentionOp tiledAttnOp,
+                        SmallVector<Operation *> &ops, RewriterBase &rewriter) {
+  Location loc = tiledAttnOp.getLoc();
+  Value keySlice = tiledAttnOp.getKey();
+  Value valueSlice = tiledAttnOp.getValue();
+  Value querySlice = tiledAttnOp.getQuery();
+  Value tiledResult = tiledAttnOp.getTiledResult();
+  Value max = tiledAttnOp.getMax();
+  Value sum = tiledAttnOp.getSum();
+
+  OpBuilder::InsertionGuard withinScfLoop(rewriter);
+  rewriter.setInsertionPointAfter(tiledAttnOp);
+  SmallVector<OpFoldResult> queryDimValues =
+      tensor::getMixedSizes(rewriter, loc, querySlice);
+  OpFoldResult headDimension = queryDimValues[1];
+  OpFoldResult sequenceTileLength = queryDimValues[0];
+
+  Type elementType = tiledAttnOp.getQueryType().getElementType();
+  auto [result, newMax, newSum] = createAttentionBody(
```

Comment:
My understanding is this is using the V1 algorithm? Now that Kunwar's patch has merged, I can try to merge V2 in https://github.com/openxla/iree/pull/14529/files which should plug right in.